pkg/operation/botanist/secrets.go

```diff
@@ -306,6 +307,32 @@ func (b *Botanist) syncShootCredentialToGarden(
 	return err
 }
 
+func (b *Botanist) deleteSSHKeypairGardenSecret(ctx context.Context) error {
```

Comment:
```suggestion
func (b *Botanist) deleteSSHKeypair(ctx context.Context) error {
```
I suggest that we keep this function name similar to the `generateSSHKeypair` func.